crates/router/src/connector/datatrans.rs

```diff
@@ -105,30 +105,30 @@ impl ConnectorCommon for Datatrans {
         res: Response,
         event_builder: Option<&mut ConnectorEvent>,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
-        let response: datatrans::DatatransResponse = res
+        let response: datatrans::DatatransErrorResponse = res
             .response
-            .parse_struct("DatatransResponse")
+            .parse_struct("DatatransErrorResponse")
             .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
 
         event_builder.map(|i| i.set_response_body(&response));
         router_env::logger::info!(connector_response=?response);
         let error_code = response
             .error
             .as_ref()
-            .and_then(|e| e.code.clone())
+            .map(|e| e.code.clone())
             .unwrap_or(consts::NO_ERROR_CODE.to_string());
         let error_message = response
             .error
             .as_ref()
-            .and_then(|e| e.message.clone())
+            .map(|e| e.message.clone())
```

Comment:
Move this `unwrap_or` to the `message` field below. And remove the `some` used for `reason`